shell/browser/electron_browser_main_parts.cc

```diff
@@ -375,10 +376,19 @@ void ElectronBrowserMainParts::ToolkitInitialized() {
   gtk_ui_delegate_ = std::make_unique<ui::GtkUiDelegateX11>(gfx::GetXDisplay());
   ui::GtkUiDelegate::SetInstance(gtk_ui_delegate_.get());
   views::LinuxUI::SetInstance(BuildGtkUi(ui::GtkUiDelegate::instance()));
-#endif
-
-#if defined(USE_AURA) && defined(USE_X11)
   views::LinuxUI::instance()->Initialize();
+
+  // GTK does not provide a way to check if current theme is dark, so we compare
+  // the text and background luminosity to get a result.
+  // This trick comes from FireFox.
+  float bg = color_utils::GetRelativeLuminance(gtk::GetBgColor("GtkLabel"));
```

Comment:
Can we observe changes to the GTK color theme somehow and do this logic on theme change?

Reply:
I have updated the code to subscribe to GTK theme changes.